### ISSUE TITLE
[BUG] Non-deterministic reads with `read_orc`

### ISSUE THREAD

**Issue by quasiben** [rapidsai/cudf#1640]
https://www.googleapis.com/download/storage/v1/b/anaconda-public-data/o/orc%2FTestOrcFile.testDate1900.orc?generation=1522611448751555&alt=media

```python
import cudf

path = 'sample.orc'
cdf = cudf.read_orc(path, skip_rows=5, num_rows=20)
print(cdf.tail().to_pandas())
```

Running the script multiple time *should* yield the same print out but it doesn't.  This also seems to only occur in new python kernel -- i.e. python `run_script.py`

cc @j-ieong

**Comment by j-ieong** [comment]
This might be the same issue we are seeing in CI, and that @OlivierNV debugged to be some uninitialized parameter...

**Comment by j-ieong** [comment]
Hrmmm, yeah looks like there's still an issue. Let me try memset everything...

**Comment by OlivierNV** [comment]
@j-ieong: are you able to get a repro locally at TOT ?
Not seeing the problem here, but my system was also not reproducing the uninit data bug that only showed up in the CI environment (definitely sounds like the uninitialized data issue, though).

**Comment by j-ieong** [comment]
@OlivierNV and I can repro it locally now.

It seems a bit weird. Repeatedly reading a single row with `skip_rows=1, num_rows=1` repros with seemingly alternating values while `skip_rows=0, num_rows=1` does not.

**Comment by OlivierNV** [comment]
I think I know what might be happening, missing a check that we're not reading the row index array above endrow (which may be less than start_row+stripe_rows), should be able to verify this shortly.

**Comment by j-ieong** [comment]
Looks like the weird values are just uninitialized or previous values in memory of the gdf_columns. Zeroing out the gdf_column memory shows that the output values are zero, suggesting that it's either not writing them out or writing them out somewhere else?

**Comment by quasiben** [comment]
@j-ieong sorry for the delayed response (was side tracked with other issues).  Yes, it was after that merge (but it looks like you are well aware of that).  I can test https://github.com/rapidsai/cudf/pull/1648 in the mean time

**Comment by harrism** [comment]
Fixed by #1648, not sure why this didn't auto-close.

**Comment by quasiben** [comment]
All working!
